packages/curve-ui-kit/src/types/util.ts

```diff
@@ -34,3 +34,9 @@ export type DeepPartial<T> = {
  * ```
  */
 export type MakeOptional<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>
+
+/**
+ * A generic type representing the result of a query operation.
+ * @template T - The type of the data returned by the query.
+ */
+export type Query<T> = { data: T | undefined; isLoading: boolean; error: Error | null | undefined }
```

Comment:
Not blocking but it feels weird we're not deriving this from the TanStack types directly, that's bound to give problems later. Freestyling here without IDE, but I'd imagine it could be something like `Pick<typeof useQuery<T>, "data" | "isLoading" | "error">`. Don't even know if `<T>` works but you get the gist I hope.